### DISCUSSION TITLE
Outline drawn to html elements after following links with `up-follow`

### DISCUSSION THREAD

**Discussion by ganchdev** [unpoly/unpoly#524] [Q&A] [answered]
I noticed that following links with the `up-follow` attribute triggers the `focus` event on fragment updates and I see an outline being drawn like so:

<img width="1014" alt="Screenshot 2023-08-31 at 23 44 00" src="https://github.com/unpoly/unpoly/assets/42203630/7813b80c-2cd2-4c6b-8911-4b73389bbe14">

And if I go back to the link:

<img width="489" alt="Screenshot 2023-08-31 at 23 44 30" src="https://github.com/unpoly/unpoly/assets/42203630/d792b98b-9e64-40cb-a48c-8ecd336299a7">

Adding `up-focus: keep` to the link removes the outline in the followed page but it still draws one to the link when going back to the index page.

Is there a way to reset the focus and never draw these outlines without doing hacks like:
```
*:focus {
  outline: none;
}
```

Thanks!

**Comment by triskweline** [comment, marked as answer]
Unpoly controls the focus for accessibility reasons. Placing the focus on new content can help make your app accessible to users with vision or motor impairments, or to users that prefer the keyboard over the mouse.

Unpoly [hides the focus outline on non-interactive elements](https://github.com/unpoly/unpoly/blob/030019974872f751949038a20c32fd1482f9a559/src/unpoly/viewport.sass#L4-L6). Since your screenshots do show outlines, I wonder if you forgot to include `unpoly.css`, or if the `outline` style is overridden by a later stylesheet further down the cascade.

**Comment by ganchdev** [comment]
> Unpoly controls the focus for accessibility reasons. Placing the focus on new content can help make your app accessible to users with vision or motor impairments, or to users that prefer the keyboard over the mouse.
> 
> Unpoly [hides the focus outline on non-interactive elements](https://github.com/unpoly/unpoly/blob/030019974872f751949038a20c32fd1482f9a559/src/unpoly/viewport.sass#L4-L6). Since your screenshots do show outlines, I wonder if you forgot to include `unpoly.css`, or if the `outline` style is overridden by a later stylesheet further down the cascade.

Yes, there was indeed something weird going on with Hugo's minification of that css file. I now got the original `unpoly.min.css` and the issue is gone.
